### PR TITLE
Support using "_p" in DSJSON parser.

### PR DESCRIPTION
**Pull request by zwd-ms**
Change the DSJSON parser to support both "p" and "_p":
- When the "_p" key is inside "c", its parsing is ignored.
- When duplicate "p" or "_p" are present, the latest one is used.

### CODE REVIEW

**Comment by olgavrou** [comment]
Do we ever have:
```
{
 "c":
     {
           "p": [...]
     }
}
```

or 

```
{
 "c":
     {
           "p": [...],
            "_p": [...]
     }
}
```
?

**Comment by zwd-ms** [comment]
I discussed this with @lokitoth. When PassthroughPDF is used and later joined by the trainer, the probability array will be copied inside "c". Currently there is not a mechanism to remove it when logging the original context.

**Comment by zwd-ms** [comment]
> Do we ever have:
> 
> ```
> {
>  "c":
>      {
>            "p": [...]
>      }
> }
> ```
> 
> or
> 
> ```
> {
>  "c":
>      {
>            "p": [...],
>             "_p": [...]
>      }
> }
> ```
> 
> ?
I discussed this with @lokitoth. When PassthroughPDF is used and later joined by the trainer, the probability array will be copied inside "c". Currently there is not a mechanism to remove it when logging the original context.

File: vowpalwabbit/parse_example_json.h [hunk 1]
```diff
@@ -852,6 +852,9 @@ class DefaultState : public BaseState<audit>
 
       if (ctx.key_length == 2 && ctx.key[1] == 'p')
       {
+        // Ignore "_p" when it is under the "c" key in decision service state.
+        if (ctx.root_state == &ctx.decision_service_state) Ignore(ctx, length);
```
Comment (jackgerrits) [inline vowpalwabbit/parse_example_json.h:856]: Could you add braces here?

**Review by jackgerrits — APPROVED** [review]

File: vowpalwabbit/parse_example_json.h (hunk 1 above)
Comment (zwd-ms) [reply vowpalwabbit/parse_example_json.h]: Done.

**Review by rajan-chari — APPROVED** [review]

**Pull request merged**